### PR TITLE
Login program

### PR DESCRIPTION
**Pull request by franciscozdo**
This PR tries to port login from NetBSD.

I have marked every part of code that uses not implemented features with `NOT_IMPLEMENTED` macro.

(Changes needed by login were done here ->  #949)

### CODE REVIEW

**Comment by franciscozdo** [comment]
**diff with original sources** (except NOT_IMPLEMENTED)

`login.c`:
```
33,43d32
< #ifndef lint
< __COPYRIGHT("@(#) Copyright (c) 1980, 1987, 1988, 1991, 1993, 1994\
<  The Regents of the University of California.  All rights reserved.");
< #endif /* not lint */
<
< #ifndef lint
< #if 0
< static char sccsid[] = "@(#)login.c   8.4 (Berkeley) 4/2/94";
< #endif
< __RCSID("$NetBSD: login.c,v 1.105 2014/11/12 22:23:38 aymeric Exp $");
< #endif /* not lint */
262c265
<       for (cnt = getdtablesize(); cnt > 2; cnt--)
---
>       for (cnt = sysconf(_SC_OPEN_MAX); cnt > 2; cnt--)
32,33d31
< __RCSID("$NetBSD: common.c,v 1.8 2015/10/28 07:27:24 shm Exp $");
<
```

`common.c`:
```
49a50
> #include <signal.h>
101a107
>     fflush(stdout);
124a131
150c159
< void __dead
---
> void
158c167
< void __dead
---
> void
165a175
376c390
< void __dead
---
> void
```
`common.h`:
```
35d34
< void   update_db(int, int, int);
40,41c39,40
< void   __dead sigint(int);
< void   __dead sleepexit(int);
---
> void   sigint(int);
> void   sleepexit(int);
43c42,43
< void   __dead timedout(int);
---
> void   timedout(int);
> #ifdef NOT_IMPLEMENTED
44a45,49
> void   update_db(int, int, int);
> #else
> #define decode_ss(...)
> #define update_db(...)
> #endif
```

File: include/pwd.h [hunk 1]
```diff
@@ -70,6 +70,10 @@
 #define _PATH_PASSWD "/etc/passwd"
 #define _PATH_MASTERPASSWD "/etc/master.passwd"
 #define _PASSWORD_LEN 128
+#define _PASSWORD_WARNDAYS 14
+#define _PASSWORD_CHGNOW                                                       \
+  -1 /* special day to force password                                          \
```
Comment (cahirwpz) [inline include/pwd.h:75]: Weird formatting - please put the comment into separate line before.

File: lib/libc/gen/execlp.c [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+
+#include <sys/cdefs.h>
+#if defined(LIBC_SCCS) && !defined(lint)
```
Comment (cahirwpz) [inline lib/libc/gen/execlp.c:33]: Please remove this cruft.

File: lib/libc/gen/ttyname.c [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #include <errno.h>
 #include <unistd.h>
 
-const char *ttyname(int fd) {
+char *ttyname(int fd) {
   /* TODO(fzdob): to implement */
```
Comment (cahirwpz) [inline lib/libc/gen/ttyname.c:5]: @j-piecuch We would need some support from you to port this function from NetBSD. Namely support for `TIOCPTSNAME`.

File: usr.bin/login/common.c [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+char *
+getloginname(void)
+{
+	int ch;
+	char *p;
+	static char nbuf[NBUFSIZ];
+
+	for (;;) {
+		(void)printf("login: ");
+    fflush(stdout);
```
Comment (cahirwpz) [inline usr.bin/login/common.c:107]: @j-piecuch Could you help @franciscozdo investigating why `fflush` is needed here, but NetBSD does without it? Maybe we're missing something in `tty` driver?

File: include/pwd.h (hunk 1 above)
Comment (franciscozdo) [reply include/pwd.h]: done

File: lib/libc/gen/execlp.c (hunk 2 above)
Comment (franciscozdo) [reply lib/libc/gen/execlp.c]: done

File: usr.bin/login/common.c (hunk 4 above)
Comment (franciscozdo) [reply usr.bin/login/common.c]: I fixed it in #962. `fstat` during `__smakebuf()` was failing

**Review by cahirwpz — APPROVED** [review]
LGTM

**Pull request merged**